### PR TITLE
fix(detailsTable): bulk select fix

### PR DESCRIPTION
**Pull request by Fewwy**
# Description

Associated Jira ticket: # (issue)
Fixing bulk select in Actions and Systems tab
https://issues.redhat.com/browse/REMEDY-283
https://issues.redhat.com/browse/REMEDY-145
https://issues.redhat.com/browse/REMEDY-90

Please include a summary of the change, what this fixes/creates/improves.
I added 2 buttons:
**Select page (number of rows)
Select all (number of values in the entire table)**
I improved the behavior of the select page buttons

**Behavior**: User has a list of things, 10 per page, 5 pages (50 items total):
First page - “Select page” --> 10 items selected
Nav to second page - “Select page” --> 20 items selected (adds to the selection)
Nav to third page - "Select page" --> 30 items selected (adds to the selection)
To unselect the third page -> "Select page" --> 20 items selected (takes out 10 - the page - from the selection)
Nav to the fourth page - “Select all” --> 50 items selected (all selected)
User chooses “Select none” --> Everything deselected, 0 items selected
User selects a single item --> 1 item selected
User goes back to page 2, chooses “Select page” --> page 2 is selected & that one item from page, 11 items total

This is a diagram for the select page button in Systems tab so you could understand it better
image

This is a diagram for the select page button in Actions tab
image

# How to test the PR
Find remediation with the name "voznesenskii test" to test systems tab.
Find remediation with the name "test-patch" to test actions tab.
Open the local dev environment, go to the /insights/remediations and find remediation with several systems
Open the table and check bulk select in the Actions tab and Activity tab.

What to check:
The dropdown should be disabled when there are no systems/actions to show. 
The dropdown should be enabled when there are systems/actions to show.
If you select the number of values < than the whole table the check mark should look like a - symbol
image

If you select the number of values = to the amount of values in the entire table the check mark should look like this 
image

# Before the change
There were no select all button

# After the change
image

# Dependent work link

# Checklist:

- [x] The commit message has the Jira ticket linked
- [x] PR has a short description
- [x] Screenshots before and after the change are added
- [ ] Tests for the changes have been added
- [ ] README.md is updated if necessary
- [ ] Needs additional dependent work

### CODE REVIEW

**Review by mkholjuraev — COMMENTED** [review]
Actions tab page selection and unselecting all need to be improved. When I select a few rows and select a page, all rows get unselected instead of selecting the whole page.

File: src/components/RemediationDetailsTable.js [hunk 1]
```diff
@@ -210,23 +213,48 @@ function RemediationDetailsTable(props) {
...
-              onClick: () => selector.props.onSelect('none'),
+              title: 'Select none (0)',
+              onClick: () => {
+                selector.props.onSelect('none');
+              },
             },
+            rows.length > 0
+              ? {
+                  title: `Select page (${rows.length})`,
+                  onClick: () => {
```
[the remark this answers is not in the document]
Comment (Fewwy) [reply src/components/RemediationDetailsTable.js:227]: I looked at the patchman UI for the example https://github.com/RedHatInsights/patchman-ui/blob/master/src/Utilities/Hooks.js#L146 
I don't see how I can make this "generic" enough to serve on the other tables. I think we have to wait for the table tools to bulk-select.

File: src/components/RemediationDetailsTable.js [hunk 2]
```diff
@@ -210,23 +213,48 @@ function RemediationDetailsTable(props) {
...
+                      ? selector.props.onSelect('page', true, 0)
+                      : rows.length === bulkSelectCheck(rows).length
+                      ? selector.props.onSelect('page', false, 0)
+                      : selector.props.onSelect('page', false, 0);
+                  },
+                }
+              : {},
+            rows.length > 0
+              ? {
+                  title: `Select all (${props.remediation.issues.length})`,
```
[the remark this answers is not in the document]
Comment (Fewwy) [reply src/components/RemediationDetailsTable.js]: Fixed

**Review by mkholjuraev — APPROVED** [review]
LGTM!

**Pull request merged**